django/contrib/gis/geos/coordseq.py

```diff
@@ -194,3 +194,28 @@ def tuple(self):
         if n == 1:
             return get_point(0)
         return tuple(get_point(i) for i in range(n))
+
+    def _is_ccw_geos(self):
+        ret = c_byte()
+        if not capi.cs_is_ccw(self.ptr, byref(ret)):
+            raise GEOSException
+        return ret.value == 1
+
+    def _is_ccw_py(self):
+        # A modified shoelace algorithm to determine polygon orientation.
+        # See https://en.wikipedia.org/wiki/Shoelace_formula.
+        area = 0.0
+        n = len(self)
+        for i in range(n):
+            j = (i + 1) % n
+            area += self[i][0] * self[j][1]
+            area -= self[j][0] * self[i][1]
+        return area > 0.0
+
+    @property
+    def is_ccw(self):
+        """
+        Return whether or not this coordinate sequence has counterclockwise
+        orientation.
+        """
+        return (self._is_ccw_py if geos_version_tuple() < (3, 7) else self._is_ccw_geos)()
```

Comment:
Is there any need for these separate methods? I could understand if it were required for testing, but that doesn't seem to be the case.

If we are to keep them separate, could we not do the following to avoid the nested method calls?
```python
is_ccw = property(_is_ccw_py if geos_version_tuple() < (3, 7) else _is_ccw_geos)
```